### ISSUE TITLE
Events are shown in wrong month

### ISSUE THREAD

**Issue by osw-mw** [lochmueller/calendarize#477]
With version 6.0.0 and a German timezone, events are sometimes shown in the wrong month in the month view.
Events on the last day of a month are shown in the calendar of the following month.

The attached patch fixes the issue by using the timezone value set by the TYPO3 core during initialization. Also, there were some mistakes in the conditions.

[IndexRepository.patch.txt](https://github.com/lochmueller/calendarize/files/5703514/IndexRepository.patch.txt)

**Comment by okmiim** [comment]
This is similar to #459 and fixed (a different way, "conditions" differ a bit) in #460.
You can find this fixed in the current version (7.3.0), however there normally "will be no new versions for old releases" (lochmueller).

_A tip for the future:_ A better way to provide patches in the future is to fork this repository and create a commit in there.
Then you can either create a pull request or crossreference the commit id in an issue.

**Comment by lochmueller** [comment]
Hey @osw-mw 
thanks for the reporting. @okmiim is right. It is much easier for us, to use a current version. I added your "code cleanup" (not the timezone) to the current master... https://github.com/lochmueller/calendarize/commit/436ecc49e0b6d95debd44e920f31df64a68f5bf7 ... and check the problem in detail later. For the timezone is also a utility function (added as comment and also as cleanup commit https://github.com/lochmueller/calendarize/commit/ce1af6c90be2ba62d214012edd914be11e1a5785 ) but we have to check the other functions too. E.g. "addDateTimeFrameConstraints" use UTC timezone...
Regards,
Tim

**Comment by osw-mw** [comment]
Hey Tim, thank you.

One additional comment – The following condition does not make sense:
```
$query->greaterThanOrEqual('endDate', $endDate),
$query->lessThan('endDate', $endDate),
```

It cannot be both smaller and bigger/equal.

It should probably be changed it to:

```
$query->lessThan('start_date', $endDate),
$query->greaterThanOrEqual('end_date', $endDate),
```
(My patch also had a wrong variable name here).

**Comment by lochmueller** [comment]
Hey @osw-mw 
thanks for the hint. This line was right in the past: https://github.com/lochmueller/calendarize/commit/554782eda23a7f9d3ce0ff02834428858864ca2b strange...
I fixed this:
https://github.com/lochmueller/calendarize/commit/c077bb40a1a2adefc2a4bc7f27d8374cceb47ed8

Just the two lines is not enough, because the " // before - in" Block select events, that start "before" and end "in" the given timeslot. So this is the right selection:
```
        $beforeIn = [
            $query->lessThan('startDate', $startDate),
            $query->greaterThanOrEqual('endDate', $startDate), // here was "$endDate" before.
            $query->lessThan('endDate', $endDate),
        ];
```

Fixed in current master.

Regards,
Tim
